spec/components/search_spec.rb

```diff
@@ -1129,6 +1129,24 @@ def search
           .to eq([post7.id, post8.id])
       end
 
+      it "can search for unlisted topics as staff" do
+        topic1 = Fabricate(:topic, visible: false)
+        Fabricate(:post, raw: 'Testing post', topic: topic1)
+        topic2 = Fabricate(:topic)
+        Fabricate(:post, raw: 'Testing post', topic: topic2)
+
+        results = Search.execute('Testing post status:unlisted', guardian: Guardian.new(Fabricate(:moderator)))
+        expect(results.posts.length).to eq(1)
```

Comment:
We need to assert for the post id here since asserting for the length only doesn't tell us if the post we expect is returned